src/runtimes/node/utils/entry_file.ts

```diff
@@ -46,14 +50,32 @@ export const conflictsWithEntryFile = (
   srcFiles: string[],
   {
     basePath,
-    mainFile,
+    extension,
+    featureFlags,
     filename,
+    mainFile,
   }: {
     basePath: string
+    extension: string
+    featureFlags: FeatureFlags
     filename: string
     mainFile: string
   },
-) => srcFiles.some((srcFile) => isNamedLikeEntryFile(srcFile, { basePath, filename }) && srcFile !== mainFile)
+) => {
+  srcFiles.forEach((srcFile) => {
+    if (featureFlags.zisi_disallow_new_entry_name && srcFile.includes(UNIQUE_ENTRY_FILE_NAME)) {
```

Comment:
Can we do the feature flag check on the outside and avoid iterating over `srcFiles` again if the flag is not empty?

Also, `srcFiles` can get quite large. Would it make sense to merge this with the `.some` so that we iterate over the array only once?

Reply:
Okay I merged them.